server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java

```diff
@@ -93,6 +97,16 @@ public void testCreateTable() throws Exception {
         refresh();
         execute("SELECT * FROM test");
         assertEquals(1L, response.rowCount());
+
+        long oid = client().admin().cluster().state(new ClusterStateRequest()).get().getState().metadata().columnOID();
+        cluster().fullRestart();
+        ClusterAdminClient clusterAdminClient = client().admin().cluster();
+        // the gateway recovery is async and
+        // it might take a bit until it reads the persisted cluster state and updates the settings expression
+        assertBusy(() -> {
+            long newOid = clusterAdminClient.state(new ClusterStateRequest()).get().getState().metadata().columnOID();
+            assertThat(oid).isEqualTo(newOid);
+        });
```

Comment:
Initially plan was to do a temporal integration test and then rewrite it to a unit test. 

Now I'm thinking maybe it worth keep it (without `@Repeat`)? Scenario is complex and maybe deserves at least 1 check in the integration test as well.

Reply:
Please create a dedicated test for this. I still think this could be an unit test but also ok with an integration test.
And the test should ensure that the oid is not 0, but that the increased value is really re-stored

Reply:
Pushed a deviated test. Asserted that oid both before and after restart is equal to concrete positive value.